io/eolearn/io/local_io.py

```diff
@@ -23,179 +63,228 @@ class ExportToTiff(SaveToDisk):
                     self.date_indices))
-            array_sub = array[np.nonzero(np.where((dates >= start_date) & (dates <= end_date), dates, 0))[0]]
-        else:
-            raise ValueError('Invalid format in {}, expected tuple or list'.format(self.date_indices))
+            return array[np.nonzero(np.where((dates >= start_date) & (dates <= end_date), dates, 0))[0]]
 
-        return array_sub
+        raise ValueError('Invalid format in {}, expected tuple or list'.format(self.date_indices))
 
-    def execute(self, eopatch, *, filename):
+    def execute(self, eopatch, *, filename=None):
+        """ Execute method
 
+        :param eopatch: input EOPatch
+        :type eopatch: EOPatch
+        :param filename: filename of tiff file or None if entire path has already been specified in `folder` parameter
+        of task initialization.
+        :type filename: str or None
+        :return: Unchanged input EOPatch
+        :rtype: EOPatch
+        """
         feature_type, feature_name = next(self.feature(eopatch))
         array = eopatch[feature_type][feature_name]
 
         array_sub = self._get_bands_subset(array)
 
-        if feature_type in [FeatureType.DATA, FeatureType.MASK, FeatureType.SCALAR]:
+        if feature_type.is_time_dependent():
             array_sub = self._get_dates_subset(array_sub, eopatch.timestamp)
-
-            if feature_type is FeatureType.SCALAR:
-                # add height and width dimensions
-                array_sub = np.expand_dims(np.expand_dims(array_sub, axis=1), axis=1)
-
         else:
             # add temporal dimension
             array_sub = np.expand_dims(array_sub, axis=0)
-            if feature_type is FeatureType.SCALAR_TIMELESS:
-                # add height and width dimensions
-                array_sub = np.expand_dims(np.expand_dims(array_sub, axis=1), axis=1)
+
+        if not feature_type.is_spatial():
+            # add height and width dimensions
+            array_sub = np.expand_dims(np.expand_dims(array_sub, axis=1), axis=1)
 
         time_dim, height, width, band_dim = array_sub.shape
 
         index = time_dim * band_dim
         dst_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
         dst_crs = {'init': CRS.ogc_string(eopatch.bbox.crs)}
 
-        # Write it out to a file.
-        with rasterio.open(os.path.join(self.folder, filename), 'w', driver='GTiff',
+        image_dtype = array_sub.dtype if self.image_dtype is None else self.image_dtype
+        if image_dtype == np.int64:
+            image_dtype = np.int32
+            warnings.warn('Data from feature {} cannot be exported to tiff with dtype numpy.int64. Will export as '
+                          'numpy.int32 instead'.format((feature_type, feature_name)))
+
+        # Write it out to a file
+        with rasterio.open(self._get_file_path(self.folder, filename), 'w', driver='GTiff',
```

Comment:
here the `self.folder` could then be omitted entirely if already taken into account before? Or do you want the additional generic folder functionality? In any case, fine by me :)

Reply:
No, there is no need for more generic functionality. Making the change.